src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -260,263 +210,323 @@ class BasePaymentsPage extends React.Component {
-            PaymentMethods.makeDefaultPaymentMethod(password, null, this.state.selectedPaymentMethod.fundID, previousPaymentMethod, currentPaymentMethod);
+    const navigateToTransferBalancePage = () => {
+        Navigation.navigate(ROUTES.SETTINGS_PAYMENTS_TRANSFER_BALANCE);
+    };
+
+    const listHeaderComponent = () => (
+        <>
+            {Permissions.canUseWallet(props.betas) && (
+                <>
+                    <View style={[styles.mv5]}>
+                        {shouldShowLoadingSpinner ? (
+                            <ActivityIndicator
+                                color={themeColors.spinner}
+                                size="large"
+                            />
+                        ) : (
+                            <OfflineWithFeedback
+                                pendingAction={CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD}
+                                errors={props.walletTerms.errors}
+                                onClose={PaymentMethods.clearWalletTermsError}
+                                errorRowStyles={[styles.ml10, styles.mr2]}
+                            >
+                                <CurrentWalletBalance />
+                            </OfflineWithFeedback>
+                        )}
+                    </View>
+                    {props.userWallet.currentBalance > 0 && (
+                        <View style={styles.mb3}>
+                            <KYCWall
+                                onSuccessfulKYC={navigateToTransferBalancePage}
+                                enablePaymentsRoute={ROUTES.SETTINGS_ENABLE_PAYMENTS}
+                                addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
+                                addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
+                                popoverPlacement="bottom"
+                            >
+                                {(triggerKYCFlow) => (
+                                    <MenuItem
+                                        title={props.translate('common.transferBalance')}
+                                        icon={Expensicons.Transfer}
+                                        onPress={triggerKYCFlow}
+                                        shouldShowRightIcon
+                                        disabled={props.network.isOffline}
+                                    />
+                                )}
+                            </KYCWall>
+                        </View>
+                    )}
+                </>
+            )}
+            <Text style={[styles.ph5, styles.textLabelSupporting, styles.mb1]}>{props.translate('paymentsPage.paymentMethodsTitle')}</Text>
+        </>
+    );
+
+    useEffect(() => {
+        PaymentMethods.openPaymentsPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!props.network.isOffline) {
+            updateShouldShowLoadingSpinner();
```

Comment:
We can use `useOnNetworkReconnect` hook and simplify this further

Reply:
Hi, @Santhosh-Sellavel 
But we need to run a little different function per isOffline.
```
    useEffect(() => {
        // If the user was previously offline, skip debouncing showing the loader
        if (!props.network.isOffline) {
            updateShouldShowLoadingSpinner();
        } else {
            debounceSetShouldShowLoadingSpinner();
        }
    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
```
As you can see, even though `debounceSetShouldShowLoadingSpinner` is the debounced function of `updateShouldShowLoadingSpinner`, they are different functions.
We need to run these different functions per network condition.
So it is some difficult to use `useOnNetworkReconnect` instead above useEffect.

Reply:
Let me check this

Reply:
Yes, `debounceSetShouldShowLoadingSpinner` will be called when network is offline.
any other callbacks doesn't depends on `props.network`.
And this effect can be triggered by only network change.